keras/src/layers/rnn/rnn.py

```diff
@@ -382,6 +382,20 @@ def call(
                 initial_state = self.get_initial_state(
                     batch_size=ops.shape(sequences)[0]
                 )
+
+        # Check for batch size mismatch when stateful
+        if self.stateful:
+            actual_batch_size = ops.shape(sequences)[0]
```

Comment:
You're using `ops.shape` here, but `.shape` for `states` you need to be consistent.

The latter returns `None` but the former might return tensors (on Tensorflow).